pkgs/development/compilers/llvm/5/compiler-rt.nix

```diff
@@ -0,0 +1,32 @@
+{ stdenv, version, fetch, cmake, python, llvm, libcxxabi }:
+with stdenv.lib;
+stdenv.mkDerivation rec {
+  name = "compiler-rt-${version}";
+  inherit version;
+  src =  fetch "compiler-rt" "0ipd4jdxpczgr2w6lzrabymz6dhzj69ywmyybjjc1q397zgrvziy";
+
+  nativeBuildInputs = [ cmake python llvm ];
+  buildInputs = stdenv.lib.optional stdenv.hostPlatform.isDarwin libcxxabi;
+
+  configureFlags = [
+    "-DCOMPILER_RT_DEFAULT_TARGET_ONLY=ON"
```

Comment:
Why default target only, just curious?

Reply:
IMO one of the big advantages of building compilers and their run-times separately is that you *can* libraries per-target. This makes it economical to support indefinite variations of targets (especially for e.g. embedded) without wasting time rebuilding the compiler itself.